train.py

```diff
@@ -1 +1,218 @@
-#TODO: MAKE TRAINING SCRIPT
+import math
+import argparse
+from data_server import load_dataset
+from time import time
+from keras.models import Sequential, load_model
+from keras.layers import Dense, Dropout, Flatten, Convolution2D, MaxPooling2D
+from keras.utils import np_utils
+from keras import optimizers
+from keras.callbacks import ModelCheckpoint, TensorBoard
+
+
+def data_preparation(args):
+    """
+    This module prepares data for training and validation process.
+
+    :param args:
+        Dataset file path.
+
+    :return:
+        6 numpy arrays containing train, test and validate datasets for x and y.
+        Shape containing height/width of processed images.
+    """
+    # Downloading and reshaping data
+    x_train, y_train, x_test, y_test, x_validate, y_validate = load_dataset(args.dataset)
+    shape = int(math.sqrt(x_train.shape[1]))
+    x_train = x_train.reshape(x_train.shape[0], shape, shape, 1)
+    x_test = x_test.reshape(x_test.shape[0], shape, shape, 1)
+    x_validate = x_validate.reshape(x_validate.shape[0], shape, shape, 1)
+
+    # Normalization
+    x_train = (x_train - 256/2) / 256
+    x_test = (x_test - 256/2) / 256
+    x_validate = (x_validate - 256/2) / 256
+
+    # Preparing a proper format of output data
+    y_train = np_utils.to_categorical(y_train, 2)
+    y_test = np_utils.to_categorical(y_test, 2)
+    y_validate = np_utils.to_categorical(y_validate, 2)
+
+    return x_train, y_train, x_test, y_test, x_validate, y_validate, shape
+
+
+def network_preparation(shape, learning_rate, saving_period, args):
+    """
+    This module prepares convolutional neural network model for future training.
+
+    :param shape:
```

Comment:
Shape is a tuple (width, height). Rather call this parameter side_length or something similar.